notificationworkerlambda/src/main/scala/com/gu/notifications/worker/delivery/apns/models/IOSMetricsRegistry.scala

```diff
@@ -0,0 +1,31 @@
+package com.gu.notifications.worker.delivery.apns.models
+
+import com.codahale.metrics.{Counter, MetricRegistry}
+import com.turo.pushy.apns.{ApnsClient, ApnsClientMetricsListener}
+
+class IOSMetricsRegistry extends ApnsClientMetricsListener {
+  val registry = new MetricRegistry()
+
+  val writeFailures: Counter = registry.counter("pushy.writeFailures")
+  val notificationSent: Counter = registry.counter("pushy.notificationSent")
+  val notificationAccepted: Counter = registry.counter("pushy.notificationAccepted")
+  val notificationRejected: Counter = registry.counter("pushy.notificationRejected")
+  val connectionAdded: Counter = registry.counter("pushy.connectionAdded")
+  val connectionRemoved: Counter = registry.counter("pushy.connectionRemoved")
+  val connectionCreationFailed: Counter = registry.counter("pushy.connectionCreationFailed")
+
+  override def handleWriteFailure(apnsClient: ApnsClient, notificationId: Long): Unit = writeFailures.inc()
+
+  override def handleNotificationSent(apnsClient: ApnsClient, notificationId: Long): Unit = notificationSent.inc()
+
+  override def handleNotificationAccepted(apnsClient: ApnsClient, notificationId: Long): Unit = notificationAccepted.inc()
```

Comment:
Do you think it would be useful to write a log when notification is accepted / sent so that we can get something like the rate of notification accepted / sent over a period of time?  Maybe we can then understand whether the Pushy is able to send notification as fast as we call Pushy API?
(Here I assume the `notificationAccepted` happens when we call Pushy API, and `notificationSent` happens when Pushy has sent the notifications to APNS.)

Anyway this PR looks great to me!

Reply:
That sounds great!

Reply:
With the extra log messages I can see the request latency in the log events:
<img width="1014" alt="Screenshot 2022-12-02 at 15 18 49" src="https://user-images.githubusercontent.com/45561419/205327052-81e46d93-0b9e-4bc6-a4be-d50c547e49e8.png">